server/rules-engine/index.js

```diff
@@ -0,0 +1,185 @@
+const DEBUG = false
+const Engine = require('json-rules-engine').Engine
+const enums = require('./enums')
+const Rule = require('json-rules-engine').Rule
+const operators = require('./operators')
+const factHandler = require('./factHandler')
+
+const _buildStandardRules = function (conditions, ruleName, eventType) {
+  const retRules = []
+  if (DEBUG) console.log('Setting up standard rules with conditions ', conditions)
+  conditions.map(condition => {
+    retRules.push(new Rule({
+      conditions: {
+        all: [condition]
+      },
+      event: {
+        type: eventType
+      },
+      name: ruleName,
+      priority: 5
+    }))
+  })
+  return retRules
+}
+
+const _setupStandardRule = function (conditions, ruleName, eventType) {
+  const rules = _buildStandardRules(conditions, ruleName, eventType)
+  rules.map(rule => {
+    this.engine.addRule(rule)
+  })
+  this.engine.on('success', function (event, almanac, ruleResult) {
+    if (event.type === eventType) {
+      almanac.addRuntimeFact(enums.ruleRejected, true)
+      almanac.factValue('ref').then(ref => {
+        if (DEBUG) console.log('Rule passed, rejecting parcel:', ref)
+      })
+    }
+  })
+}
+
+const _buildAcceptedItemsRules = function () {
+  return [new Rule({
+    conditions: {
+      all: [{
+        fact: enums.ruleRejected,
+        operator: 'notEqual',
+        value: true
+      }]
+    },
+    event: {
+      type: enums.acceptedEventName
+    },
+    priority: 1,
+    name: 'matchAcceptedItems'
+  })]
+}
+
+const _setupAcceptedItemsRule = function (funcToCall) {
+  const rules = _buildAcceptedItemsRules()
+  rules.map(rule => {
+    this.engine.addRule(rule)
+  })
+  this.engine.on('success', function (event, almanac, ruleResult) {
+    if (event.type === enums.acceptedEventName) {
+      if (typeof funcToCall !== 'undefined') {
+        funcToCall(event, almanac, ruleResult)
+      }
+    } else if (event.type === enums.rejectedFactAdder) {
+      // almanac.addRuntimeFact(enums.ruleRejected, false)
+    }
```

Comment:
Could this last else if be deleted along with the comment?

Reply:
Yes, no longer needed.